### PR TITLE
[Backport 2.x] Change password security message (#3057)

### PR DESCRIPTION
**Pull request by derek-ho**
Backports 847f911 from #3057

Returns the error message instead of the validation message because validation message should already have been shown in above case statements.

Signed-off-by: Craig Perkins <cwperx@amazon.com>
Signed-off-by: Derek Ho <dxho@amazon.com>
Co-authored-by: Craig Perkins <cwperx@amazon.com>
(cherry picked from commit 847f91108a4a10c8368fde26a8f79266edf423c3)

### CODE REVIEW

File: src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java [hunk 1]
```diff
@@ -86,19 +86,31 @@ ErrorType validate(final String username, final String password) {
...
+            return RequestContentValidator.ValidationError.INVALID_PASSWORD_TOO_SHORT;
+>>>>>>> 847f9110 (Change password security message (#3057))
         }
         if (password.length() > MAX_LENGTH) {
             logger.debug(
                 "Password is too long, the maximum required length is {}, but current length is {}",
                 MAX_LENGTH,
                 password.length()
             );
+<<<<<<< HEAD
```
Comment (willyborankin) [inline src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java:101]: a leftover after rebase of something. it wont compile :-)

File: src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java [hunk 2]
```diff
@@ -86,19 +86,31 @@ ErrorType validate(final String username, final String password) {
                 minPasswordLength,
                 password.length()
             );
+<<<<<<< HEAD
```
Comment (willyborankin) [inline src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java:89]: a leftover after rebase of something. it wont compile :-)

File: src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java [hunk 3]
```diff
@@ -86,19 +86,31 @@ ErrorType validate(final String username, final String password) {
...
                 MAX_LENGTH,
                 password.length()
             );
+<<<<<<< HEAD
             return ErrorType.INVALID_PASSWORD;
         }
         if (Objects.nonNull(passwordRegexpPattern) && !passwordRegexpPattern.matcher(password).matches()) {
             logger.debug("Regex does not match password");
             return ErrorType.INVALID_PASSWORD;
+=======
```
Comment (willyborankin) [inline src/main/java/org/opensearch/security/dlic/rest/validation/PasswordValidator.java:107]: a leftover after rebase of something. it wont compile :-)

File: src/test/java/org/opensearch/security/dlic/rest/validation/PasswordValidatorTest.java [hunk 4]
```diff
@@ -126,12 +126,21 @@ public void testRegExpBasedValidation() {
                 .put(SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, "(?=.*[A-Z])(?=.*[^a-zA-Z\\\\d])(?=.*[0-9])(?=.*[a-z]).{8,}")
                 .build()
         );
+<<<<<<< HEAD
```
Comment (willyborankin) [inline src/test/java/org/opensearch/security/dlic/rest/validation/PasswordValidatorTest.java:129]: a leftover after rebase of something. it wont compile :-)

**Comment by peternied** [comment]
@derek-ho I've updated the description and the title of the issue to capture its 'backportness' Looks like there are still CI failures, can you look into them?

**Comment by peternied** [comment]
@derek-ho Still looks broken, happy to take another look after CI is passing.  Let us know if you need a hand.

**Comment by derek-ho** [comment]
@peternied should be fixed now, sorry. It was blocked on backported API changes, should have called that out to avoid any confusion.

**Review by peternied — APPROVED** [review]

**Pull request merged**